### PR TITLE
Whitelisting Player Feature with Merkel Root

### PR DESCRIPTION
**Pull request by viraj124**
Enhancement

- https://goodghosting.atlassian.net/browse/SC-8?atlOrigin=eyJpIjoiNGRkOTI3NTc4NGI5NDcyMTg0NDVhYTJhODY4N2QwNWUiLCJwIjoiaiJ9

### CODE REVIEW

File: contracts/GoodGhosting.sol [hunk 1]
```diff
@@ -106,8 +106,9 @@ contract GoodGhosting is Ownable, Pausable {
         uint256 _segmentPayment,
         uint256 _earlyWithdrawalFee,
         uint256 _customFee,
-        address _dataProvider
-    ) public {
+        address _dataProvider,
```
Comment (fodisi) [inline contracts/GoodGhosting.sol:109]: Do we use this data provider param for anything?

**Review by fodisi — APPROVED** [review]

File: contracts/GoodGhosting.sol (hunk 1 above)
Comment (viraj124) [reply contracts/GoodGhosting.sol:109]: using it here @fodisi https://github.com/Good-Ghosting/goodghosting-smart-contracts/blob/feature/whitelist-players/contracts/GoodGhosting.sol#L132

Comment (fodisi) [reply contracts/GoodGhosting.sol:109]: Oh, I see. Sorry. I was reviewing this on my cellphone and it looked like it was a new parameter. than I checked for any new assignment to it and didn't see it, so I posted the question above. Totally my bad - sorry about it.

**Comment by viraj124** [comment]
@fodisi merging this discussed with @RachBLondon

**Pull request merged**